### PR TITLE
fix(web): handle edge cases in timeToSeconds function to prevent crashes

### LINKED ISSUE #20583: Cannot read properties of undefined (reading 'split')
### I have searched the existing issues, both open and closed, to make sure this is not a duplicate report.

- [x] Yes

### The bug

```
Uncaught TypeError: Cannot read properties of undefined (reading 'split')
    at Xr (B1LP_BB7.js:1:16803)
    at Bl2m2zqA.js:1:9030
    at Vt (sL99BBiK.js:1:12693)
    at Xe (sL99BBiK.js:1:5659)
    at wt (sL99BBiK.js:1:5708)
    at se (sL99BBiK.js:1:15102)
    at get durationInSeconds (Bl2m2zqA.js:1:9186)
    at pt (DyJJRVrM.js:1:2242)
    at Ge (Bl2m2zqA.js:1:1241)
    at ut (Bl2m2zqA.js:1:9068)
```

Perhaps related to this `/api/timeline/bucket?timeBucket=2025-08-01T00%3A00%3A00.000Z&visibility=timeline&withPartners=true&withStacked=true` response:

```json
{
    "city": [
        null,
        null,
        null,
        null,
        null,
        null,
        "Berga",
        null,
        "Berga",
        "Berga",
        "Berga",
        "Berga",
        null,
        "Berga",
        "Berga",
        "Berga",
        "Berga",
        "Berga",
        "Berga",
        "Berga",
        "Berga",
        "Berga",
        "Berga",
        "Berga",
        "Berga",
        "Berga",
        null
    ],
    "country": [
        null,
        null,
        null,
        null,
        null,
        null,
        "Germany",
        null,
        "Germany",
        "Germany",
        "Germany",
        "Germany",
        null,
        "Germany",
        "Germany",
        "Germany",
        "Germany",
        "Germany",
        "Germany",
        "Germany",
        "Germany",
        "Germany",
        "Germany",
        "Germany",
        "Germany",
        "Germany",
        null
    ],
    "duration": [
        "00:03:00.096",
        "00:03:00.096",
        "00:03:00.096",
        "00:03:00.096",
        "00:03:00.096",
        "00:03:00.096",
        "00:00:16.579",
        "0",
        null,
        "00:00:29.952",
        null,
        "00:00:35.953",
        "00:00:37.571",
        "00:00:14.281",
        "00:00:36.446",
        "00:00:08.694",
        "00:02:00.395",
        "00:00:12.719",
        "00:00:11.647",
        "00:00:13.804",
        "00:01:39.368",
        "00:00:03.467",
        "00:00:17.097",
        "00:00:19.010",
        "00:00:20.315",
        "00:00:35.999",
        "00:00:53.855"
    ],
    "id": [
        "61f0bc53-b0e4-4d6c-a5fd-fe072afffbcc",
        "10ac1444-ff1e-459d-8b29-7817ad8922f4",
        "a1b8ef5b-1bda-4ef1-a0b4-1d519b31cdc1",
        "7f608b7e-4492-43b1-ad63-b838e8c12c10",
        "0e138e24-6e14-4b1d-9455-be36cc3f867e",
        "91162332-56fc-48d4-86d3-1f2654577d86",
        "48dbfbe0-d4bf-4118-9356-2c15fd5081a9",
        "66aa965f-79b3-4c9a-89e1-d7fe426001c6",
        "a07e4edb-f992-4390-a927-81528b7f61cd",
        "403f6a1c-861a-46fc-bf32-a0ee7f049bf9",
        "f864a354-57fa-4d76-8766-a9ed0cb15df0",
        "59a0445a-9b95-41e9-826c-f322a393b78e",
        "8a2c2621-393c-4870-bb25-f822becfdb6c",
        "51ecaecf-5250-4040-9108-6ca693ce002a",
        "3de3effa-22ad-47fc-b09a-a921a6ce0de1",
        "dbbf7e54-25a7-446a-9caf-db6b53b42fac",
        "966d5b62-f712-4b42-9417-fe42191416c2",
        "8fbd32f8-efe9-481b-91c6-8fb44ed0346d",
        "bd644162-9c37-4581-aec0-8a9f42a99588",
        "b9afdef8-9c52-47d3-9189-e1a454088a87",
        "9aa28ccd-9862-45a2-8736-0690e486b849",
        "5954f43e-510e-41a6-93ec-ee228f02c05a",
        "05ef6248-d8d8-4321-b592-ce5dcc5db095",
        "c8900d86-5c36-4fb2-aca6-ba5775c25f61",
        "0274bceb-fd50-42c9-8acf-348775571ea1",
        "aed5b99d-ef42-4905-8d80-19e44adefe9f",
        "36221a42-5513-45e2-853f-54021ae9e472"
    ],
    "visibility": [
        "timeline",
        "timeline",
        "timeline",
        "timeline",
        "timeline",
        "timeline",
        "timeline",
        "timeline",
        "timeline",
        "timeline",
        "timeline",
        "timeline",
        "timeline",
        "timeline",
        "timeline",
        "timeline",
        "timeline",
        "timeline",
        "timeline",
        "timeline",
        "timeline",
        "timeline",
        "timeline",
        "timeline",
        "timeline",
        "timeline",
        "timeline"
    ],
    "isFavorite": [
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        false
    ],
    "isImage": [
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        true,
        false,
        true,
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        false
    ],
    "isTrashed": [
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        false
    ],
    "livePhotoVideoId": [
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null
    ],
    "fileCreatedAt": [
        "2025-08-02T16:42:54",
        "2025-08-02T06:09:29",
        "2025-08-02T02:24:59",
        "2025-08-01T22:22:49",
        "2025-08-01T22:04:51",
        "2025-08-01T22:01:27",
        "2025-08-01T18:11:38",
        "2025-08-01T17:37:28.328",
        "2025-08-01T17:37:11.97",
        "2025-08-01T12:32:18",
        "2025-08-01T11:38:00.936",
        "2025-08-01T11:21:41",
        "2025-08-01T11:13:17",
        "2025-08-01T10:00:47",
        "2025-08-01T09:59:24",
        "2025-08-01T09:57:49",
        "2025-08-01T09:50:45",
        "2025-08-01T09:48:41",
        "2025-08-01T09:48:17",
        "2025-08-01T09:28:05",
        "2025-08-01T09:27:34",
        "2025-08-01T09:25:28",
        "2025-08-01T09:24:34",
        "2025-08-01T09:24:14",
        "2025-08-01T09:19:04",
        "2025-08-01T09:18:19",
        "2025-08-01T09:16:37"
    ],
    "localOffsetHours": [
        0,
        0,
        0,
        0,
        0,
        0,
        2,
        0,
        2,
        2,
        2,
        2,
        0,
        2,
        2,
        2,
        2,
        2,
        2,
        2,
        2,
        2,
        2,
        2,
        2,
        2,
        0
    ],
    "ownerId": [
        "ad2055bf-3ce4-4bc0-9884-87707fa0ee04",
        "ad2055bf-3ce4-4bc0-9884-87707fa0ee04",
        "ad2055bf-3ce4-4bc0-9884-87707fa0ee04",
        "ad2055bf-3ce4-4bc0-9884-87707fa0ee04",
        "ad2055bf-3ce4-4bc0-9884-87707fa0ee04",
        "ad2055bf-3ce4-4bc0-9884-87707fa0ee04",
        "ad2055bf-3ce4-4bc0-9884-87707fa0ee04",
        "ad2055bf-3ce4-4bc0-9884-87707fa0ee04",
        "ad2055bf-3ce4-4bc0-9884-87707fa0ee04",
        "ad2055bf-3ce4-4bc0-9884-87707fa0ee04",
        "ad2055bf-3ce4-4bc0-9884-87707fa0ee04",
        "ad2055bf-3ce4-4bc0-9884-87707fa0ee04",
        "ad2055bf-3ce4-4bc0-9884-87707fa0ee04",
        "ad2055bf-3ce4-4bc0-9884-87707fa0ee04",
        "ad2055bf-3ce4-4bc0-9884-87707fa0ee04",
        "ad2055bf-3ce4-4bc0-9884-87707fa0ee04",
        "ad2055bf-3ce4-4bc0-9884-87707fa0ee04",
        "ad2055bf-3ce4-4bc0-9884-87707fa0ee04",
        "ad2055bf-3ce4-4bc0-9884-87707fa0ee04",
        "ad2055bf-3ce4-4bc0-9884-87707fa0ee04",
        "ad2055bf-3ce4-4bc0-9884-87707fa0ee04",
        "ad2055bf-3ce4-4bc0-9884-87707fa0ee04",
        "ad2055bf-3ce4-4bc0-9884-87707fa0ee04",
        "ad2055bf-3ce4-4bc0-9884-87707fa0ee04",
        "ad2055bf-3ce4-4bc0-9884-87707fa0ee04",
        "ad2055bf-3ce4-4bc0-9884-87707fa0ee04",
        "ad2055bf-3ce4-4bc0-9884-87707fa0ee04"
    ],
    "projectionType": [
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null
    ],
    "ratio": [
        1.778,
        1.778,
        1.778,
        1.778,
        1.778,
        1.778,
        0.563,
        1,
        0.563,
        0.563,
        1.778,
        0.563,
        1.778,
        0.563,
        1.778,
        0.563,
        1.778,
        1.778,
        1.778,
        0.563,
        0.563,
        0.563,
        0.563,
        0.563,
        1.778,
        1.778,
        0.563
    ],
    "status": [
        "active",
        "active",
        "active",
        "active",
        "active",
        "active",
        "active",
        "active",
        "active",
        "active",
        "active",
        "active",
        "active",
        "active",
        "active",
        "active",
        "active",
        "active",
        "active",
        "active",
        "active",
        "active",
        "active",
        "active",
        "active",
        "active",
        "active"
    ],
    "thumbhash": [
        "3ucJDIJbdmn+iIh7lmhfqEUIBQ==",
        "4OcJFIJ6d4jfiJeWd2dZmUAHFg==",
        "zwcSBIA0eYegeYh2eHjKb1sDSQ==",
        "zAcOBIAIlXiCeYhjrInYf3kCaA==",
        "zgcOBIAbhndkeIhgjIqxrO80BQ==",
        "xwcKBIA0iYegendpiHivPwkFbA==",
        "kfgNDABG+HdniJNYeYgHQpPzCA==",
        null,
        "HwgGDADa+Km2mIp6ZoirYBYgSw==",
        "IvgJDAKW+Ia3dnmXl4f9iXTxFg==",
        "lvgJFIIHvkZPxomKdmXpkMQKpA==",
        "oNkJLAKGCIe4eIVYmJiWgKz4FA==",
        "WskFHIJ2iXeAiIeLeHaFoPwFhA==",
        "3sgBDALEJZylmHCpO8ZwtfEX4w==",
        "GOkFDIK6nYiviKmMm6aG7g+aRw==",
        "2ukFJARVOXj5Z2mG2bh1gK34CQ==",
        "1ugFDIJ5dFn+WWaPdndR4u8LGg==",
        "1ugFDIKKZYr9emafd3hg0f8KCg==",
        "1+gFDIJ5ZYr8eleviIhQwN4qCw==",
        "F/gJDARiZXsHp1qJeUj0rHsLKQ==",
        "X+gFBAaIWFYHuVeH62Wsj4cHSA==",
        "nugFDAZYVYcKeXZ5yFa/VmgGWQ==",
        "YPgBHAQJpk3Gh6gZS+pqr1sHFg==",
        "XPgFDAKSuY5mVmC1palnb54IJg==",
        "WQgGDIJ5vNxPc0eAbJe6X5sPaA==",
        "IfgBDISWyQpXi51gg4ZgXwdjVw==",
        "2ckFJAL4md6Kz71MWZiWkHb81Q=="
    ],
    "stack": [
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null
    ]
}
```

### The OS that Immich Server is running on

docker

### Version of Immich Server

1.137.3

### Version of Immich Mobile App

unrelated

### Platform with the issue

- [ ] Server
- [x] Web
- [ ] Mobile

### Your docker-compose.yml content

```YAML
unrelated
```

### Your .env content

```Shell
unrelated
```

### Reproduction steps

1. Upload some asset causing the bug (?)
2. See timeline no longer loading

### Relevant log output

```shell
None
```

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by baldarn**
## Description

- Add proper validation for null, undefined, empty string, and '0' inputs
- Prevent split() calls on malformed time strings
- Validate parsed numbers to handle non-numeric values
- Return 0 for invalid inputs instead of throwing errors
- Add comprehensive test coverage for edge cases

Fixes #20583 - 'Cannot read properties of undefined (reading split)' error when timeline receives assets with invalid duration values

Fixes # (issue)

## How Has This Been Tested?

```web/src/lib/utils/date-time.spec.ts```

- handles "0" string input
- handles empty string input
- handles malformed time strings
- handles time strings with invalid numbers

<details><summary><h2>Screenshots (if appropriate)</h2></summary>

</details>

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation if applicable
- [ ] I have no unrelated changes in the PR.
- [ ] I have confirmed that any new dependencies are strictly necessary.
- [x] I have written tests for new code (if applicable)
- [x] I have followed naming conventions/patterns in the surrounding code
- [ ] All code in `src/services/` uses repositories implementations for database calls, filesystem operations, etc.
- [ ] All code in `src/repositories/` is pretty basic/simple and does not have any immich specific logic (that belongs in `src/services/`)

### CODE REVIEW

File: web/src/lib/utils/date-time.ts [hunk 1]
```diff
@@ -7,11 +7,30 @@ import { get } from 'svelte/store';
...
+  if (parts[2]) {
+    parts[2] = parts[2].split('.').slice(0, 2).join('.');
+  }
 
   const [hours, minutes, seconds] = parts.map(Number);
 
+  // Validate that the parts are valid numbers
+  if (isNaN(hours) || isNaN(minutes) || isNaN(seconds)) {
+    return 0;
+  }
```
Comment (danieldietzler) [inline web/src/lib/utils/date-time.ts]: Can't we just rely on Luxon for this? https://moment.github.io/luxon/api-docs/index.html#duration; `Duration.fromISOTime(time)`

Comment (baldarn) [reply web/src/lib/utils/date-time.ts]: I implemented Luxon `fromISOTime` and left the other validations, do you think this is too much?

Comment (danieldietzler) [reply web/src/lib/utils/date-time.ts]: I don't think that validation is necessary at all. Luxon will just error if the provided time is bad.

File: web/src/lib/utils/date-time.ts [hunk 2]
```diff
@@ -7,14 +7,16 @@ import { get } from 'svelte/store';
  * Convert time like `01:02:03.456` to seconds.
  */
 export function timeToSeconds(time: string) {
-  const parts = time.split(':');
-  parts[2] = parts[2].split('.').slice(0, 2).join('.');
+  // Handle edge cases: null, undefined, empty string, or "0"
```
Comment (danieldietzler) [inline web/src/lib/utils/date-time.ts]:
```suggestion
```

File: web/src/lib/utils/date-time.ts [hunk 3]
```diff
@@ -7,14 +7,16 @@ import { get } from 'svelte/store';
...
+  if (!time || time === '0') {
+    return 0;
+  }
 
-  const [hours, minutes, seconds] = parts.map(Number);
+  const seconds = Duration.fromISOTime(time).as('seconds');
 
-  return Duration.fromObject({ hours, minutes, seconds }).as('seconds');
+  // Return 0 for invalid inputs that result in NaN
```
Comment (danieldietzler) [inline web/src/lib/utils/date-time.ts]:
```suggestion
```

**Review by danieldietzler — APPROVED** [review]
There's no need to comment (more or less) trivial pieces of code. Same applies for the tests, I'd remove the comments there too. Besides this looks good though, thanks!

**Review by danieldietzler — APPROVED** [review]

**Review by danieldietzler — APPROVED** [review]
